client-js/test/spec/tsTestSpec.ts

```diff
@@ -0,0 +1,61 @@
+import {Agent} from '../test/helloWorld';
+import {Observable} from '../app/scripts/observable';
+import {Lifetime} from '../app/scripts/lifetime';
+
+
+describe("Hello world", function() {
+  var agent;
+
+  beforeEach(function() {
+    agent = new Agent();
+  });
+
+  it("says hello", function() {
+    expect(agent.helloWorld()).toEqual("Hello world!");
+  });
+});
+
+/*
+* Test to create new observable and add callback
+* This is mostly to see if the plumbing for the
+* tests interfaces with the scripts in /app
+*/
+
+describe("observable", function() {
```

Comment:
looks ok; personally I prefer to have things close together, rather than doing setup in a function like `beforeEach`, which spreads out the code that you have to read to see what a test is doing. Maybe if you have lots of specs that share common setup this could make sense, but even then, I would prefer to have a function that you call explicitly within the specs instead of relying on side effects from `beforeEach`. For example:
```javascript
function setup(): Observable<any> {
  var observable = new Observable<any>();
  // do a bunch of setup
  return observable;
}

it("does something", function(done) {
  var observable = setup();
  ...
}
```

For the particular case here, I would say it's better to do the work like calling the observable in the spec functions, not the setup function. The setup should be reserved for instantiating objects, etc., not performing the actual logic that you want to test.

```javascript
it("calls added callbacks asynchronously", function(done) {
  var observable = new Observable<string>();
  observable.add((msg) => {
    expect(msg).toEqual("foo");
    done();
  });
  observable.call("foo");
});
```

Anyway, we'll have to see how our usage evolves. For now, what you have is fine.